packages/cicero-core/src/contract.js

```diff
@@ -26,6 +31,138 @@ const TemplateInstance = require('./templateinstance.js');
  */
 class Contract extends TemplateInstance {
 
+    /**
+     * Create the Clause and link it to a Template.
+     * @param {Template} template  - the template for the clause
+     */
+     constructor(template) {
+        super(template);
+    }
+
+     /**
+     * Sign Instance
+     * @param {string} contractText - contract text extracted from contract markdown
+     * @param {object} signatureObject - contains signatures if existing parties who signed the contract. null if no one hasn't signed.
+     * @param {string} keyStorePath - path of the keystore to be used
+     * @param {string} keyStorePassword - password for the keystore file
+     * @return {object} object conatining array of all signatures
+     */
+      signInstance(contractText, signatureObject, keyStorePath, keyStorePassword) {
+        const ciceroMarkTransformer = new CiceroMarkTransformer();
+        const dom = ciceroMarkTransformer.fromMarkdownCicero( contractText, 'json' );
+        const resultText = this.formatCiceroMark(dom);
```

Comment:
Unclear to me why this roundtrip, or whether this is the right kind of roundtrip. Why not just hash on `data` + TemplateMark for instance? Can we hash on JSON directly?